cpp/src/partitioning/partitioning.cu

```diff
@@ -724,6 +724,35 @@ struct dispatch_map_type {
 
 namespace detail {
 namespace {
+
+/**
+ * @brief  This hash function simply returns the value that is asked to be hash
+ * reinterpreted as the result_type of the functor.
```

Comment:
This sentence is missing something to be grammatically correct. Maybe we can simplify it like:
```suggestion
 * @brief  This hash function simply returns the input value cast to the
 * result_type of the functor.
```